util/list.c

```diff
@@ -95,7 +95,7 @@ struct _list_desc {
 /** @brief Default function used to compare element in the list ( \ref f_cmp) */
 static int32_t default_comparator(void *data1, void *data2)
 {
-	return (int32_t)(data1 - data2);
+	return (int32_t)((int32_t *)data1 - (int32_t *)data2);
```

Comment:
The original code here is weird. @mchindri can you please explain what this default comparator should do ? If it's comparing pointers that point to data of unknown size then I think you should drop this default comparator - it can lead to weird behaviour. I saw it's not even used.

Reply:
@mphalke I think it's ok for now to cast to int32_t * even though the default comparator doesn't make sense (to me). At least it unblocks building with IAR.

Reply:
The list implementation make it possible to work as an ordered list. In order to short the items a comparator function is needed. If no comparator is provided, then this default_comparator function is used. My best though was to compare the addresses because anything else can be compared (the data content is unknown).
In this sense, I think the cast used is not relevant so this change seems to be valid.